commit 9fc39494173d67f4625c3534debfedf81db589fd (refs/changes/97/2297/1)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-07-28T11:30:03-07:00 (2 years, 2 months ago)
    
    IMGAPI-637 correctly report when docker image is update to date

diff --git a/CHANGES.md b/CHANGES.md
index 8860f44..2b0e9fd 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,9 @@
 # IMGAPI changelog
 
+## 4.0.3
+
+- IMGAPI-637: correctly report when docker image is update to date
+
 ## 4.0.2
 
 - IMGAPI-636: docker pull retry not working for v2.1 images
diff --git a/lib/docker.js b/lib/docker.js
index 7e43df3..5394a3b 100644
--- a/lib/docker.js
+++ b/lib/docker.js
@@ -20,6 +20,7 @@ var zlib = require('zlib');
 var assert = require('assert-plus');
 var drc = require('docker-registry-client');
 var imgmanifest = require('imgmanifest');
+var jsprim = require('jsprim');
 var once = require('once');
 var streampeek = require('buffer-peek-stream');
 var vasync = require('vasync');
@@ -1282,7 +1283,7 @@ function _dockerV22Pull(ctx, cb) {
                 }
             });
 
-            var status = (ctx.newFileInfoFromUuid.length === 0
+            var status = (jsprim.isEmpty(ctx.newFileInfoFromUuid)
                 ? 'Status: Image is up to date for ' + ctx.repoAndRef
                 : 'Status: Downloaded newer image for ' + ctx.repoAndRef);
             resMessage({
diff --git a/package.json b/package.json
index 2832144..c220bb5 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "4.0.2",
+  "version": "4.0.3",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
@@ -20,8 +20,8 @@
     "handlebars": "4.0.5",
     "imgmanifest": "3.0.0",
     "json": "9.0.4",
+    "jsprim": "^1.4.0",
     "ldap-filter": "0.3.1",
-    "uuid": "2.0.2",
     "manta": "3.1.2",
     "manta-dir-watcher": "1.1.0",
     "manta-sync": "0.4.2",
@@ -40,6 +40,7 @@
     "sshpk": "1.13.0",
     "trace-event": "1.2.0",
     "ufds": "1.2.0",
+    "uuid": "2.0.2",
     "vasync": "1.6.3",
     "verror": "1.6.0",
     "wf-client": "0.2.1"
